src/main/java/app/linguistai/bmvp/service/gamification/StoreService.java

```diff
@@ -81,6 +85,59 @@ public RStoreItems getAllEnabledStoreItems(int page, int size) throws SomethingW
         }
     }
 
+    @Transactional(readOnly = true)
+    public RQuizItems getAllQuizItems(String email, int page, int size) throws Exception {
+        try {
+            User user = accountRepository.findUserByEmail(email).orElseThrow(() -> new NotFoundException(User.class.getSimpleName(), true));
+
+            Pageable pageable = PageRequest.of(page, size);
+
+            Page<StoreItem> storeItemsPage = storeItemRepository.findByTypeIn(QUIZ_TYPES, pageable);
+
+            // Map store items to user items, creating items with 0 quantity if the user does not have the item
+            List<RUserItem> userItems = storeItemsPage.getContent().stream()
```

Comment:
This part is hard to read even with the descriptive comments (perhaps because of the nesting), can you refactor this code or even better, is it possible to move this functionality to its own private method?

Reply:
Yes, I moved it to a private method and refactored it a little bit to use guard clauses.